[BUG] Compilation error when the git branch name contains special characters

### Bug summary

The CI reports an error in `deepmd_version.h` when the branch name contains special characters: `bump-scikit-build-core-to->=0.5,<0.7`.

Maybe a quote should be added?

### DeePMD-kit Version

devel

### TensorFlow Version

2.14.0

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

See https://github.com/njzjz/deepmd-kit/actions/runs/6658998734/job/18097120044

```
     FAILED: lmp/plugin/CMakeFiles/deepmd_lmpplugin.dir/__/pair_deepmd.cpp.o
    /opt/rh/gcc-toolset-11/root/usr/bin/c++ -DGOOGLE_CUDA -DLAMMPS_VERSION_NUMBER=20230802 -DLMPPLUGIN -D_GLIBCXX_USE_CXX11_ABI=0 -Ddeepmd_lmpplugin_EXPORTS -I/project/build/py37-none-manylinux_2_28_x86_64/lmp/plugin -I/project/source/lmp/plugin/.. -I/project/build/py37-none-manylinux_2_28_x86_64/_deps/lammps_download-src/src/PLUGIN -I/project/build/py37-none-manylinux_2_28_x86_64/_deps/lammps_download-src/src/KSPACE -I/project/build/py37-none-manylinux_2_28_x86_64/_deps/lammps_download-src/src -I/project/build/py37-none-manylinux_2_28_x86_64/_deps/lammps_download-src/src/EXTRA-FIX -I/project/source/api_c/include -I/usr/include/mpich-x86_64 -fopenmp -O3 -DNDEBUG -fPIC -std=gnu++17 -Winvalid-pch -include /project/build/py37-none-manylinux_2_28_x86_64/lmp/plugin/CMakeFiles/deepmd_lmpplugin.dir/cmake_pch.hxx -MD -MT lmp/plugin/CMakeFiles/deepmd_lmpplugin.dir/__/pair_deepmd.cpp.o -MF lmp/plugin/CMakeFiles/deepmd_lmpplugin.dir/__/pair_deepmd.cpp.o.d -o lmp/plugin/CMakeFiles/deepmd_lmpplugin.dir/__/pair_deepmd.cpp.o -c /project/source/lmp/pair_deepmd.cpp
    /project/source/lmp/pair_deepmd.cpp:441:46: error: macro "DPMD_CVT_STR" passed 2 arguments, but takes just 1
      441 |     cout << pre << "source branch:      " << STR_GIT_BRANCH << endl;
          |                                              ^~~~~~~~~~~~~~
    In file included from /project/source/lmp/pair_deepmd.cpp:31:
    /project/build/py37-none-manylinux_2_28_x86_64/lmp/plugin/deepmd_version.h:8: note: macro "DPMD_CVT_STR" defined here
        8 | #define DPMD_CVT_STR(x) #x
          |
    /project/source/lmp/pair_deepmd.cpp: In member function ‘void LAMMPS_NS::PairDeepMD::print_summary(std::string) const’:
    /project/build/py37-none-manylinux_2_28_x86_64/lmp/plugin/deepmd_version.h:9:27: error: ‘DPMD_CVT_STR’ was not declared in this scope
        9 | #define DPMD_CVT_ASSTR(X) DPMD_CVT_STR(X)
          |                           ^~~~~~~~~~~~
    /project/build/py37-none-manylinux_2_28_x86_64/lmp/plugin/deepmd_version.h:12:24: note: in expansion of macro ‘DPMD_CVT_ASSTR’
       12 | #define STR_GIT_BRANCH DPMD_CVT_ASSTR(GIT_BRANCH)
          |                        ^~~~~~~~~~~~~~
    /project/source/lmp/pair_deepmd.cpp:441:46: note: in expansion of macro ‘STR_GIT_BRANCH’
      441 |     cout << pre << "source branch:      " << STR_GIT_BRANCH << endl;
          |                                              ^~~~~~~~~~~~~~
```

### Steps to Reproduce

Checkout a branch with special characters.

### Further Information, Files, and Links

_No response_

--- a/source/lmp/deepmd_version.h.in
+++ b/source/lmp/deepmd_version.h.in
@@ -5,7 +5,7 @@
 #define DEEPMD_ROOT @CMAKE_INSTALL_PREFIX@
 #define TensorFlow_INCLUDE_DIRS @TensorFlow_INCLUDE_DIRS@
 #define TensorFlow_LIBRARY @TensorFlow_LIBRARY@
-#define DPMD_CVT_STR(x) #x
+#define DPMD_CVT_STR(...) #__VA_ARGS__
 #define DPMD_CVT_ASSTR(X) DPMD_CVT_STR(X)
 #define STR_GIT_SUMM DPMD_CVT_ASSTR(GIT_SUMM)
 #define STR_GIT_HASH DPMD_CVT_ASSTR(GIT_HASH)

